File: frontend/types/organization.ts

import { JobLocation } from "./job";
export interface Organization {
  id: string
  name: string
  logo_url: string
  industry: string
  tier: string
  is_mock: boolean
  languages: string[]
  created_at: string
  size_range: string
  updated_at: string
  description: string
  website_url: string
  company_type: string
  founded_year: number
  mock_batch_id: string | null
  cover_image_url: string
  primary_location: JobLocation
  verification_status: string
  additional_locations: JobLocation[]
}

export enum OrganizationMemberRole {
  OWNER = "OWNER",
  ADMIN = "ADMIN",
  MEMBER = "MEMBER",
  GUEST = "GUEST",
}

export enum OrganizationMemberStatus {
  ACTIVE = "ACTIVE",
  PENDING = "PENDING",
  INACTIVE = "INACTIVE",
}

export interface OrganizationMember {
  id: string
  organization_id: string
  user_id: string
  role: OrganizationMemberRole
  status: OrganizationMemberStatus
  email: string
  created_at: string
  updated_at: string
} 

File: frontend/app/types/organization.py
from pydantic import BaseModel
from typing import List, Optional, Literal, Dict
from datetime import datetime

class LocalizedText(BaseModel):
    en: str
    fr: Optional[str] = None

class LocalizedLocation(BaseModel):
    city: LocalizedText
    state: LocalizedText
    country: LocalizedText
    postal_code: LocalizedText

class Organization(BaseModel):
    id: str
    name: LocalizedText
    tier: Literal["FREE", "PREMIUM", "ENTERPRISE"]
    is_mock: bool
    industry: str
    logo_url: str
    languages: List[str]
    created_at: datetime
    size_range: str
    updated_at: datetime
    description: LocalizedText
    website_url: str
    company_type: Literal["CORPORATION", "STARTUP", "NONPROFIT", "GOVERNMENT", "OTHER"]
    founded_year: int
    mock_batch_id: Optional[str] = None
    cover_image_url: str
    primary_location: LocalizedLocation
    verification_status: Literal["PENDING", "VERIFIED", "REJECTED"]
    additional_locations: List[LocalizedLocation] = []

class OrganizationsResponse(BaseModel):
    organizations: Dict[str, Organization] 